velox/dwio/parquet/reader/StructColumnReader.cpp

```diff
@@ -92,12 +92,36 @@ void StructColumnReader::read(
   SelectiveStructColumnReader::read(offset, rows, nullptr);
 }
 
-void StructColumnReader::enqueueRowGroup(
+std::shared_ptr<dwio::common::BufferedInput>
+StructColumnReader::enqueueRowGroup(
+    uint32_t index,
+    const std::shared_ptr<dwio::common::BufferedInput>& input) {
+  // Check whether the entire row group has already loaded. If true,
+  // reuse the input and data will be read from buffer.
+  // Or else create a new input.
+  auto newInput = isRowGroupBuffered(index, *input) ? input : input->clone();
```

Comment:
Can simplify as
```
 if(isRowGroupBuffered(index, *input)) {
   return input;
 }
 auto newInput = input->clone();
 enqueueRowGroup(index, *newInput);
 newInput->load(dwio::common::LogType::STRIPE);
 return newInput;
```

Reply:
Still need to call `enqueueRowGroup` if row group is buffered. It assgins stream_ of the column chunk to the buffered data.
https://github.com/facebookincubator/velox/blob/78c6d45c4ddcb97fcbac30d76fcafaf040b39556/velox/dwio/parquet/reader/ParquetData.cpp#L109